packages/api-client/src/user/UserAPI.ts

```diff
@@ -533,12 +533,40 @@ export class UserAPI {
       method: 'post',
       url: UserAPI.URL.LIST_USERS,
     };
-
-    const {data: userData} = await this.client.sendJSON<User[] | UsersReponse>(config);
-    if (isUsersResponse(userData)) {
-      return userData;
+    try {
+      const {data: userData} = await this.client.sendJSON<User[] | UsersReponse>(config);
+      if (isUsersResponse(userData)) {
+        return userData;
+      }
+      return {found: userData};
+    } catch (error: any) {
+      if (
+        error.label === BackendErrorLabel.FEDERATION_NOT_AVAILABLE ||
+        error.label === BackendErrorLabel.FEDERATION_BACKEND_NOT_FOUND ||
+        error.label === BackendErrorLabel.FEDERATION_REMOTE_ERROR ||
+        error.label === BackendErrorLabel.FEDERATION_TLS_ERROR
+      ) {
+        const selfDomain = this.backendFeatures.domain;
```

Comment:
How about a comment that explains what happens there ;) ?

Reply:
```suggestion
      ) {
        /* UserAPI,backendFeatures.domain returns the domain of the self user*/
        const selfDomain = this.backendFeatures.domain;
```
something like this or more detail on what `UserAPI.backendFeature` is?

Reply:
No I mean a general comments that explain how we handle federation errors (sorry I didn't mean to comment on this line in particular 😅 )